Hast du greeter.c vergessen hinzuzufügen? make schlägt mit folgender Fehlermeldung fehl:
~~~
mkdir -p obj
gcc -c -o obj/main.o src/main.c -I./include
make: *** No rule to make target 'obj/greeter.o', needed by 'tcpServer'.  Stop.
~~~

--- a/src/greeter.c
+++ b/src/greeter.c
@@ -1,0 +1,8 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+char* greet(char *name) {
+    char *greeting = malloc(128);
+    sprintf(greeting, "Hello, %s!", name);
+    return greeting;
+}

